build-source-tarball.sh

```diff
@@ -27,11 +27,11 @@ if [ $SKIP_BUILD -ne 1 ]; then
         rm -rf "$SCRIPT_ROOT/bin"
     fi
 
-    $SCRIPT_ROOT/clean-submodules.sh
+    $SCRIPT_ROOT/clean.sh -a
     $SCRIPT_ROOT/build.sh /p:ArchiveDownloadedPackages=true /flp:v=detailed
 fi
 
-$SCRIPT_ROOT/clean-submodules.sh
+$SCRIPT_ROOT/clean.sh -a
```

Comment:
-a doesn't seem correct, because later in the script, the Tools dir is copied to $TARBALL_ROOT and -a removes it.

Reply:
You're right, I think this breaks with `--skip-build` on.  I used `-a` because clean.sh no longer has this behavior with `-a` off:
```
git clean -dxf "$SCRIPT_ROOT/src/javascriptservices/"
git clean -dxf "$SCRIPT_ROOT/src/netcorecli-fsc/"
git clean -dxf "$SCRIPT_ROOT/src/reference-assemblies/"
```
It looks like we do want that behavior, as later we copy from these directories as well.  I might need to add an option to clean.sh - say `-k` for clean-well-known?  @weshaggard

Reply:
I would prefer you just fix the Clean target in build.proj the correctly clean these directories instead of adding another option to the clean script.

Reply:
Ok, I'll take a look at that.

Reply:
Ideally those sub-directories wouldn't be dropping files in the source tree and instead the paths of their outputs should be reset to be under the bin folder in the root and then it would bet cleaned when you clean the bin folder. If for some reason that cannot be done then doing a git clean on those sub-dirs might be acceptable until those issues can be addressed.